include/kokkos-utils/view/slice.hpp

```diff
@@ -0,0 +1,78 @@
+#ifndef KOKKOS_UTILS_VIEW_SLICE_HPP
+#define KOKKOS_UTILS_VIEW_SLICE_HPP
+
+#include "kokkos-utils/concepts/View.hpp"
+
+/**
+ * @file
+ *
+ * This file is a collection of helper functions similar to what can be found in
+ * the standard @c mdspan header, related to retrieving slices of a @c Kokkos::View.
+ *
+ * References:
+ *  - https://en.cppreference.com/w/cpp/container/mdspan
+ *  - https://www.open-std.org/jtc1/sc22/wg21/docs/papers/2023/p2630r4.html
+ */
+
+namespace Kokkos::utils::view
+{
+
+namespace impl
+{
+
+template <typename ViewType, typename TupleA, size_t... A, typename... Indices>
+KOKKOS_FUNCTION
+constexpr auto slice(ViewType&& view, TupleA&& alls, std::index_sequence<A...>, Indices&&... indices) {
+    return Kokkos::subview(
+        std::forward<ViewType>(view),
+        std::forward<Indices>(indices)...,
+        std::get<A>(std::forward<TupleA>(alls))...
+    );
+}
+
+} // namespace impl
+
+/**
+ * @brief Get a subview, given the first @p indices. The rest is filled with @c Kokkos::ALL.
+ *
+ * @note Though similar to @c std::submdspan, we think it it OK not to provide the remaining
+ *       slice specifiers if they are all @c Kokkos::ALL iif the user specifies what is the expected rank
+ *       of the input view. This ensures that the user will not face unexpected behaviors.
+ *       Therefore, we allow the following:
+ *       @code
+ *       auto sliced = slice<4>(view_of_rank_4, 1, 2);
+ *       @endcode
+ *       which is strictly equivalent to:
+ *       @code
+ *       auto sliced = subview(view_of_rank_4, 1, 2, Kokkos::ALL, Kokkos::ALL);
+ *       @endcode
+ */
+template <size_t Rank, typename ViewType, typename... Indices>
+requires concepts::ViewOfRank<std::remove_reference_t<ViewType>, Rank>
+KOKKOS_FUNCTION
+constexpr auto slice(ViewType&& view, Indices&&... indices)
+{
+    constexpr auto size = Rank - sizeof...(Indices);
+    return impl::slice(
+        std::forward<ViewType>(view),
+        std::array<Kokkos::ALL_t, size>{},
+        std::make_index_sequence<size>{},
+        std::forward<Indices>(indices)...
+    );
+}
+
+//! @overload
```

Comment:
OK. That's nice how the two functions work together.

Reply:
:+1: